apps/src/aichat/chatApi.ts

```diff
@@ -0,0 +1,70 @@
+import {Role, Status, ChatCompletionMessage} from './types';
+import HttpClient from '@cdo/apps/util/HttpClient';
+import {CHAT_COMPLETION_URL} from './constants';
+
+/**
+ * This function sends a POST request to the chat completion backend controller.
+ */
+async function postOpenaiChatCompletion(
+  messagesToSend: OpenaiChatCompletionMessage[]
+): Promise<OpenaiChatCompletionMessage | null> {
+  const payload = {messages: messagesToSend};
+  const response = await HttpClient.post(
+    CHAT_COMPLETION_URL,
+    JSON.stringify(payload),
+    true,
+    {
+      'Content-Type': 'application/json; charset=UTF-8',
+    }
+  );
+  if (response.ok) {
+    return await response.json();
+  } else {
+    return null;
+  }
+}
+
+const formatForChatCompletion = (
+  chatMessages: ChatCompletionMessage[]
+): OpenaiChatCompletionMessage[] => {
+  return chatMessages.map(message => {
+    return {role: message.role, content: message.chatMessageText};
+  });
+};
+
+/**
+ * This function formats chat completion messages including the system prompt, passes them
+ * to `postOpenaiChatCompletion`, then returns the status of the response and assistant message if successful.
+ */
+export async function getChatCompletionMessage(
+  systemPrompt: string,
+  userMessageId: number,
+  newMessage: string,
+  chatMessages: ChatCompletionMessage[]
+): Promise<ChatCompletionResponse> {
+  const messagesToSend = [
+    {role: Role.SYSTEM, content: systemPrompt},
+    ...formatForChatCompletion(chatMessages),
+    {role: Role.USER, content: newMessage},
+  ];
+
+  const response = await postOpenaiChatCompletion(messagesToSend);
```

Comment:
Oh one thing I forgot - we should wrap this in a `try`/`catch` since the HttpClient will throw on any errors. For now we could just console log, or we could go ahead and report an error using `Lab2MetricsReporter`, but it would be good to at least have some error handling here.

Reply:
Thanks! Added a `try/catch`. In terms of conveying error to view - will add a ticket for this.
Just adding notes: once PII and content filters are in place, if the user message is too personal (`Status.PERSONAL`), the message will not be displayed and instead a warning with a yellow background will show up.
If inappropriate (`Status.INAPPROPRIATE`), a warning with a red background will be displayed.
But if the post request just fails due to a timeout or if chat assistant response is inappropriate (I think we'll eventually monitor this) - need to decide how to handle these cases.